### DISCUSSION TITLE
facing issues with 8812au on nvidia Nano

### DISCUSSION THREAD

**Discussion by Abu-Abdullah** [svpcom/wfb-ng#123] [Questions] [answered]
first of all thank you very much for this lib and the continuous support here.

I have tried it with raspberry Pi 3 (gs/drone) and it is working fine. 

Because i'm not able to get the max out of Pi camera encoder to check the max bandwidth i can get from 8812au, i'm trying now to test it with gs (PC ubuntu 20.04) and drone (Nano ubuntu 18.04)

Now installation on PC/ubuntu 20.04 is fine and the tunnel was establish correctly but i couldn't get Nano to work:
```
Mar 14 11:19:00 oracle-desktop wfb-server[7261]: 2021-03-14 11:19:00+0400 [-] Log opened.
Mar 14 11:19:00 oracle-desktop wfb-server[7261]: 2021-03-14 11:19:00+0400 [-] # iw reg set BO
Mar 14 11:19:00 oracle-desktop wfb-server[7261]: 2021-03-14 11:19:00+0400 [-] # ifconfig wlan0 down
Mar 14 11:19:00 oracle-desktop wfb-server[7261]: 2021-03-14 11:19:00+0400 [-] command failed: Operation not supported (-95)
Mar 14 11:19:00 oracle-desktop wfb-server[7261]: 2021-03-14 11:19:00+0400 [-] Stopping reactor due to fatal error: RC 161: iw dev wlan0 set monitor otherbss
```
it shows that the card is not supporting injection i guess. but the same card is just working fine with other boards (PC/Pi) so i guess the issue is with my configuration on Nano. few details:
```
oracle@oracle-desktop:~$ ethtool -i wlan0
driver: rtl8812au
version:
firmware-version:
expansion-rom-version:
bus-info: 1-2.1:1.0
supports-statistics: no
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no
```

```
oracle@oracle-desktop:~$ find /lib/modules | grep -i 88xx
/lib/modules/4.9.201-tegra/updates/dkms/88XXau.ko
```
wlan0 is disabled as well in NetworkManager
No output from `ps uaxwwww | grep wlan0` as well

the output of the `iw list` is showing that the card is not supporting monitor mode
```
Supported interface modes:
                 * IBSS
                 * managed
                 * AP
                 * P2P-client
                 * P2P-GO
                 * P2P-device
```
any advice on how to continue is appreciated

**Comment by svpcom** [comment, marked as answer]
```
sha1sum /lib/modules/4.9.201-tegra/updates/dkms/88XXau.ko
sha1sum /path/to/your/compiled/88XXau.ko
```

**Reply by svpcom** [reply]
If checksums are the same, the try:
```
rmmod 88XXau
insmod /path/to/your/compiled/88XXau.ko
```

**Reply by svpcom** [reply]
It seems that you load default driver. Patched driver should be **rtl88XXau**:
```
$ ethtool -i wlan1
driver: rtl88XXau
version: 
firmware-version: 
expansion-rom-version: 
bus-info: 3-2:1.0
supports-statistics: no
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no
```
Try ``lsmod  | grep 8812`` and rmmod it

**Reply by Abu-Abdullah** [reply]
thank you @svpcom , indeed the checksums are different. i didn't search correctly, the default driver is loaded. i just remove it `/lib/modules/4.9.201-tegra/kernel/drivers/net/wireless/realtek/rtl8812au/rtl8812au.ko` and it started as expected now
